coordinator/core/core.go

```diff
@@ -301,7 +327,15 @@ func (c *Core) generateCert(dnsNames []string) (*x509.Certificate, *ecdsa.Privat
 		IsCA:                  true,
 	}
 
-	certRaw, err := x509.CreateCertificate(rand.Reader, &template, &template, &privk.PublicKey, privk)
+	var certRaw []byte
+	if parentCertificate == nil || parentPrivateKey == nil {
+		certRaw, err = x509.CreateCertificate(rand.Reader, &template, &template, &privk.PublicKey, privk)
+	} else {
+		template.Issuer = parentCertificate.Issuer
+		template.Subject.CommonName = CoordinatorIntermediateName
+		certRaw, err = x509.CreateCertificate(rand.Reader, &template, parentCertificate, &privk.PublicKey, parentPrivateKey)
+	}
```

Comment:
```suggestion
	if parentCertificate == nil {
		parentCertificate = &template
		parentPrivateKey = privk
	}
	certRaw, err := x509.CreateCertificate(rand.Reader, &template, parentCertificate, &privk.PublicKey, parentPrivateKey)
```
make behavior of this function more explicit by passing commonName as a parameter to generateCert
CreateCertificate does not use Issuer